influxql/iterator.gen.go.tmpl

```diff
@@ -167,6 +164,18 @@ func (itr *{{$k.name}}MergeIterator) Close() error {
 
 // Next returns the next point from the iterator.
 func (itr *{{$k.name}}MergeIterator) Next() *{{$k.Name}}Point {
+	if !itr.init {
+		items := itr.heap.items
+		itr.heap.items = make([]*{{$k.name}}MergeHeapItem, 0, len(items))
+		for _, item := range items {
+			if item.itr.peek() == nil {
+				continue
+			}
+			itr.heap.items = append(itr.heap.items, item)
+		}
+		heap.Init(itr.heap)
+	}
+
```

Comment:
From the commit message where I added this commit (duplicated here in case of a rebase):

> The MergeIterator creation function would call `peek()` on the iterator
> to initialize the heap. Since this function can sometimes take a long
> time (such as a huge aggregate query on a shard), the
> `influxql.Select()` wouldn't return until the query had already been
> completed.
>
> The `influxql.Select()` call should be just the creation of the
> iterators and shouldn't calculate anything. This is important for future
> features like the point limiter that have to be initialized after the
> `influxql.Select()` call.